timescaledb/how-to-guides/hypertables/hypertables-and-unique-indexes.md

```diff
@@ -0,0 +1,140 @@
+# Hypertable partitioning and unique indexes
+Hypertable partitioning introduces some restrictions on unique indexes.
+
+## Partitioning columns
+To determine what unique indexes you can create on a hypertable, first
+identify your partitioning columns.
+
+In TimescaleDB, the following columns are partitioning columns:
+*   The time column used to create the hypertable. Every TimescaleDB hypertable
+    is partitioned by time.
+*   Any space-partitioning columns. Space partitions are optional and not
+    included in every hypertable. You have a space-partitioning column if you
+    specify a `partitioning_column` parameter when you call
+    [`create_hypertable`][create_hypertable] or
+    [`create_distributed_hypertable`][create_distributed_hypertable].
+
+## Partitioning and unique indexes
+When you create a unique index on a hypertable, it must contain all the
+partitioning columns of the hypertable. It may contain other columns as well,
+and they may be arranged in any order.
+
+This restriction is necessary to guarantee global uniqueness in the index.
+<!--TODO: PR for link destination to be merged
+For more information, see the section on [indexes in hypertables][local-indexes].
+-->
+
+If you don't include all the partitioning columns, you get this error:
+```
+ ERROR: cannot create a unique index without the column "<COLUMN_NAME>" (used in
+partitioning) 
+```
+
+<highlight type="note">
+In PostgreSQL, a primary key is a unique index with a `NOT NULL` constraint.
+Any information about unique indexes also applies to primary keys.
+</highlight>
+
+### Create unique indexes on a hypertable
+Create a unique index as you normally would in PostgreSQL, using `CREATE UNIQUE
+INDEX`. Make sure to include all partitioning columns in the index.
+
+For example, say you create a hypertable partitioned on `time` and
+`device_id`:
+```sql
+CREATE TABLE hypertable_example(
+  time TIMESTAMPTZ,
+  user_id BIGINT,
+  device_id BIGINT,
+  value FLOAT
+);
+
+SELECT * FROM create_hypertable(
+  'hypertable_example',
+  'time',
+  partitioning_column => 'device_id',
+  number_partitions => 4
+);
+```
+
+You can create a unique index on `time` and `device_id`:
+```sql
+CREATE UNIQUE INDEX idx_time_deviceid
+ON hypertable_example(time, device_id);
+```
+
+You can create a unique index on `time`, `user_id`, and `device_id`:
+```sql
+CREATE UNIQUE INDEX idx_time_userid_deviceid
+ON hypertable_example(time, user_id, device_id);
+```
+
+But you cannot create a unique index without `device_id`, because `device_id` is
+a partitioning column. This does not work:
+```sql
+-- This gives you an error
+CREATE UNIQUE INDEX idx_time
```

Comment:
I'd do this as a unique index on, say device_id, or user_id, device_id both because they're more realistic, and because I think it's important to demonstrate that the time column must be included in any unique index.